src/argilla/client/workspaces.py

```diff
@@ -14,67 +13,238 @@
+                    " make sure that the ID you provided is a valid one. Otherwise,"
+                    " you can create a new one via the `Workspace.create` method."
+                ) from e
+            except ValidationApiError as e:
+                raise ValueError(
+                    "The ID you provided is not a valid UUID, so please make sure"
+                    " that the ID you provided is a valid one."
+                ) from e
+            except BaseClientError as e:
+                raise RuntimeError(f"Error while retrieving workspace with id=`{id}` from Argilla.") from e
+        elif name is not None:
+            try:
+                workspaces = workspaces_api.list_workspaces(active_client().http_client.httpx).parsed
+            except Exception as e:
+                raise RuntimeError("Error while retrieving the list of workspaces from Argilla.") from e
+            for workspace in workspaces:
+                if workspace.name == name:
+                    existing_workspace = workspace
+                    break
+
+            if existing_workspace is None:
+                raise ValueError(
+                    f"Workspace with name=`{name}` doesn't exist in Argilla, so please"
+                    "create it via the `Workspace.create` method as follows:"
+                    f" `Workspace(name=`{name}`)`."
+                )
+        self.__dict__.update(existing_workspace.dict())
+
+    @property
+    def users(self) -> List["WorkspaceUserModel"]:
+        """Returns the list of users linked to the workspace.
+
+        Returns:
+            A list of `WorkspaceUserModel` instances.
+        """
+        if not hasattr(self, "__users") or self.__users is None:
+            self.__users = workspaces_api.list_workspace_users(active_client().http_client.httpx, self.id).parsed
+        return self.__users
+
+    def __str__(self) -> str:
+        return f"Workspace(id={self.id}, name={self.name}, users={self.users}, insterted_at={self.inserted_at}, updated_at={self.updated_at}"
+
+    def add_user(self, id: str) -> None:
+        """Adds an existing user to the workspace in Argilla.
 
         Args:
-            name: the name of the workspace to be managed. If not provided, the current
-                workspace will be used.
-            id: the ID of the workspace to be managed.
+            id: the ID of the user to be added to the workspace. The user must exist in Argilla.
+
+        Raises:
+            ValueError: if the user with the provided ID already exists in the workspace.
+            RuntimeError: if there was an error while adding the user to the workspace.
+
+        Examples:
+            >>> from argilla import rg
+            >>> workspace = rg.Workspace("my-workspace")
+            >>> workspace.add_user("my-user")
         """
-        self.name = name
-        self.id = id or None
+        try:
+            created_user = workspaces_api.create_workspace_user(
```

Comment:
you're creating a workspace-user relationship, right?

Reply:
Yes, I just renamed the method to be `add_user` in the `Workspace` class just to avoid using the `create_workspace_user` naming as I found it a little bit confusing, while `add_user` seems more straight forward